### PR TITLE
RawShaderMaterial: added support for custom defines

### PR DESCRIPTION
**Pull request by WestLangley**
As suggested in #9119.

Also added `SHADER_NAME`.

### CODE REVIEW

File: src/renderers/webgl/WebGLProgram.js [hunk 1]
```diff
@@ -282,8 +282,21 @@ THREE.WebGLProgram = ( function () {
 
 		if ( material instanceof THREE.RawShaderMaterial ) {
 
-			prefixVertex = '';
-			prefixFragment = '';
+			prefixVertex = [
+
+				'#define SHADER_NAME ' + material.__webglShader.name,
```
Comment (mrdoob) [inline src/renderers/webgl/WebGLProgram.js]: Uh, where does that `.name` come from?

Comment (WestLangley) [inline src/renderers/webgl/WebGLProgram.js]: @mrdoob Well, I remembered the conversation... Took awhile to find it again, though... : - )

https://gist.github.com/spite/7ae92212b4f28076ba29

And your commit https://github.com/mrdoob/three.js/commit/b428a19e5064bf2ba7506706cd41495ffaadcdf9

Comment (mrdoob) [inline src/renderers/webgl/WebGLProgram.js]: > And your commit https://github.com/mrdoob/three.js/commit/b428a19e5064bf2ba7506706cd41495ffaadcdf9

Oh, right right. I thought that name was user submitted.

Comment (mrdoob) [inline src/renderers/webgl/WebGLProgram.js]: Looks good then!

**Comment by mrdoob** [comment]
Thanks!

**Comment by mrdoob** [comment]
Thanks!

File: src/renderers/webgl/WebGLProgram.js (hunk 1 above)
Comment (mrdoob) [inline src/renderers/webgl/WebGLProgram.js]: Hmm, wouldn't it be simpler to just do this?

```js
'#define SHADER_NAME RawShaderMaterial',
```

Comment (WestLangley) [inline src/renderers/webgl/WebGLProgram.js]: I think being consistent with the coding pattern used for the other materials has benefits, but it is up to you.

**Comment by mrdoob** [comment]
Actually, [one example](https://github.com/mrdoob/three.js/blob/master/examples/webgl_interactive_instances_gpu.html#L142) is writing its own `SHADER_NAME` and now glsl is crashing because the define is being redefined... So I'll have to revert that part.

**Pull request merged**